cloud/cluster/memory/subscription.go

```diff
@@ -0,0 +1,56 @@
+package memory
+
+import (
+	"github.com/scootdev/scoot/cloud/cluster"
+)
+
+// subscriber promptly reads its input and maintains a queue so that the sender
+// doesn't have to worry about a send blocking.
+type subscriber struct {
+	inCh    chan []cluster.NodeUpdate
+	outCh   chan []cluster.NodeUpdate
+	closeCh chan interface{}
+	queue   []cluster.NodeUpdate
+}
+
+func makeSubscription(initial []cluster.Node, closeCh chan interface{}, inCh chan []cluster.NodeUpdate) cluster.Subscription {
+	s := &subscriber{
+		inCh:    inCh,
+		outCh:   make(chan []cluster.NodeUpdate),
+		closeCh: closeCh,
+		queue:   nil,
+	}
+	go s.loop()
```

Comment:
Why's that?

I think it's intentional. That's why there's no way to kill a goroutine in Go (because killing a thread in Java was never correct). It's like memory, yeah, with garbage collection, there's no way to know it's reclaimed, but that's intentional.

The cost of a goroutine is much lower than the cost of a thread in another language. So I think we should think of it like orphaning memory: bad, worth doing right, important to fix if it starts killing tasks, but not something to be anxious about.